### PR TITLE
test: manual local http server runner

### PR DESCRIPTION
**Pull request by sadym-chromium**
Add script to run local http server, so that it can be run manually.

Bug #575

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,7 +270,25 @@ Use the `PORT` environment variable to connect to another port:
...
+### Local http server
+
+E2e tests use local http
+server [`pytest-httpserver`](https://pytest-httpserver.readthedocs.io/), which is run
+automatically with the tests. However,
+sometimes it is useful to run the http server outside the test
+case, for example for manual debugging. This can be done by running:
+
+```sh
+python tools/run_local_http_server.py
```
Comment (thiagowfx) [inline README.md:282]:
```suggestion
python3 tools/run_local_http_server.py
```

File: tests/conftest.py [hunk 2]
```diff
@@ -25,6 +25,16 @@
 from tools.local_http_server import LocalHttpServer
 
 
+@pytest_asyncio.fixture
```
Comment (thiagowfx) [inline tests/conftest.py:28]: defining the same fixture twice

File: README.md [hunk 3]
```diff
@@ -270,7 +270,25 @@ Use the `PORT` environment variable to connect to another port:
 PORT=8081 npm run e2e
 ```
 
-#### Examples
+### Local http server
+
+E2e tests use local http
```
Comment (thiagowfx) [inline README.md:275]:
```suggestion
E2E tests use local http
```

File: tools/run_local_http_server.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+import sys
+
+from pytest_httpserver import HTTPServer
+
+# Current directory is noit a module, so to import `LocalHttpServer`, it's path
```
Comment (thiagowfx) [inline tools/run_local_http_server.py:21]:
```suggestion
# Current directory is not a module, so to import `LocalHttpServer`, its path
```

File: tools/run_local_http_server.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# has to be added to `sys.path`. It is done relative to this file's directory.
+# The `flake8` is disabled for this reason.
+sys.path.append(os.path.dirname(__file__) + '/../tests/tools/')
```
Comment (thiagowfx) [inline tools/run_local_http_server.py:24]: Please use the pathlib library here (`from pathlib import Path`) to append to paths

File: tools/run_local_http_server.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import local_http_server  # noqa: E402
+
+httpserver = HTTPServer()
+httpserver.start()
+local_http_server = local_http_server.LocalHttpServer(httpserver)
+print("qweqwe Local http server started...")
```
Comment (thiagowfx) [inline tools/run_local_http_server.py:31]: delete qweqwe

**Review by thiagowfx — CHANGES_REQUESTED** [review]
.

File: tests/conftest.py (hunk 2 above)
Comment (sadym-chromium) [reply tests/conftest.py:28]: Indeed. I screw up with rebasing... Fixed.

File: tools/run_local_http_server.py (hunk 5 above)
Comment (sadym-chromium) [reply tools/run_local_http_server.py:24]: done

File: tools/run_local_http_server.py (hunk 6 above)
Comment (sadym-chromium) [reply tools/run_local_http_server.py:31]: oops )

File: tools/run_local_http_server.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+import sys
+from pathlib import Path
+
+from pytest_httpserver import HTTPServer
+
+# Current directory is not a module, so to import `LocalHttpServer`, its path
+# has to be added to `sys.path`. It is done relative to this file's directory.
+# The `flake8` is disabled for this reason.
+sys.path.append(str(Path(__file__).resolve().parent / '../tests/tools/'))
```
Comment (thiagowfx) [inline tools/run_local_http_server.py:24]: Are you sure you need `..` if you added `parent`? Or are you looking for `.parent.parent`?

File: tools/run_local_http_server.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+import local_http_server  # noqa: E402
+
+httpserver = HTTPServer()
+httpserver.start()
+local_http_server = local_http_server.LocalHttpServer(httpserver)
+print("Local http server started...")
```
Comment (thiagowfx) [inline tools/run_local_http_server.py:31]: Just a slightly improved suggestion:

```
print(f"""
Local http server started...
  - 200: {...}
  - 301 / permanent redirect: {...}
"""

And maybe also print in the first line where the server started? E.g. localhost:8080 or similar.
```

**Review by thiagowfx — APPROVED** [review]
LGTM. Giving approval as the requested changes are straightforward but happy to take another look if you want

File: tools/run_local_http_server.py (hunk 7 above)
Comment (sadym-chromium) [reply tools/run_local_http_server.py:24]: `parent` refers to the file's directory. `tools` can be referenced either by `.parents[1]`, or by `../` prefix. Do you have a strong preference of one over another way?

**Review by thiagowfx — APPROVED** [review]

File: tools/run_local_http_server.py (hunk 7 above)
Comment (thiagowfx) [reply tools/run_local_http_server.py]: The appeal of the pathlib library is to be OS agnostic, so `.parent` is always better than `../` - IIRC the latter does not work on Windows. LGTM as it is now.

File: tools/run_local_http_server.py (hunk 8 above)
Comment (sadym-chromium) [reply tools/run_local_http_server.py]: Write where the server runs requires adding it in the API, and I don't think it is needed for tests. The developer can easily see the address part in the lines below.

**Review by thiagowfx — APPROVED** [review]

**Pull request merged**